tests/platform_tests/api/test_watchdog.py

```diff
@@ -47,57 +50,78 @@ def conf(self, request, duthost):
 
         if platform in test_config and 'default' in test_config[platform]:
             config.update(test_config[platform]['default'])
-
         if platform in test_config and hwsku in test_config[platform]:
             config.update(test_config[platform][hwsku])
 
-        assert 'valid_timeout' in config
+        pytest_assert('valid_timeout' in config, "valid_timeout is not defined in config")
         # make sure watchdog won't reboot the system when test sleeps for @TEST_WAIT_TIME_SECONDS
-        assert config['valid_timeout'] > TEST_WAIT_TIME_SECONDS * 2
-
+        pytest_assert(config['valid_timeout'] > TEST_WAIT_TIME_SECONDS * 2, "valid_timeout {} seconds is too short".format(config['valid_timeout']))
         return config
 
-
+    @pytest.mark.dependency()
     def test_arm_disarm_states(self, duthost, localhost, platform_api_conn, conf):
         ''' arm watchdog with a valid timeout value, verify it is in armed state,
         disarm watchdog and verify it is in disarmed state
         '''
         watchdog_timeout = conf['valid_timeout']
         actual_timeout = watchdog.arm(platform_api_conn, watchdog_timeout)
 
-        assert actual_timeout != -1
-        assert actual_timeout >= watchdog_timeout
-        assert watchdog.is_armed(platform_api_conn)
+        if self.expect(actual_timeout is not None, "Watchdog.arm is not supported"):
+            pytest_assert(isinstance(actual_timeout, int), "actual_timeout appears incorrect")
```

Comment:
Based on our conversation above, should this become another nested `if self.expect(...)` call?

Reply:
good catch!